### PR TITLE
[JAVA] [Doc] Improve java doc for PG

### PR DESCRIPTION
**Pull request by clay4megtr**
## Why are these changes needed?

Improve java doc for Placement Group.

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by raulchen — COMMENTED** [review]
Can you add the  demo code `java/test/src/main/java/io/ray/docdemo/`, so we can make sure the demo code is always correct.

**Comment by raulchen** [comment]
Currently, there are too many static PG-related methods in `Ray.java`. It'd be better to organize them in a separate class, e.g., `PlacementGroups.create()`.

**Comment by clay4444** [comment]
> Currently, there are too many static PG-related methods in `Ray.java`. It'd be better to organize them in a separate class, e.g., `PlacementGroups.create()`.

Yeah, actually I have implemented this in this pr: https://github.com/ray-project/ray/pull/14729

File: doc/source/placement-group.rst [hunk 1]
```diff
@@ -128,77 +171,196 @@ Let's create a placement group. Recall that each bundle is a collection of resou
...
+        # Wait until placement group is created.
+        ray.get(pg.ready())
 
-    # Wait until placement group is created.
-    ray.get(pg.ready())
+    .. group-tab:: Java
+
+      .. code-block:: java
+
+        // Before you run this code, you need to make sure there are two "CPU" and two "extra_resource" are available.
```
Comment (raulchen) [inline doc/source/placement-group.rst:263]: we can also init a cluster with these resources in Java
set system property ray.head-args

**Review by raulchen — COMMENTED** [review]
Please also add this:

Java demo code in this documentation can be found `here <https://github.com/ray-project/ray/blob/master/java/test/src/main/java/io/ray/docdemo/PlacementGroupDemo.java>`__.

**Review by raulchen — APPROVED** [review]

File: doc/source/placement-group.rst [hunk 2]
```diff
@@ -35,40 +35,83 @@ A **placement group strategy** is an algorithm for selecting nodes for bundle pl
 Starting a placement group
 --------------------------
 
-Ray placement group can be created via the ``ray.util.placement_group`` API. Placement groups take in a list of bundles and a :ref:`placement strategy <pgroup-strategy>`:
+Ray placement group can be created via the ``ray.util.placement_group`` API if you are using Python. With Java, you can use ``Ray.createPlacementGroup`` API. Placement groups take in a list of bundles and a :ref:`placement strategy <pgroup-strategy>`:
```
[the remark this answers is not in the document]
Comment (clay4444) [reply doc/source/placement-group.rst]: updated.

File: doc/source/placement-group.rst [hunk 3]
```diff
@@ -35,40 +35,83 @@ A **placement group strategy** is an algorithm for selecting nodes for bundle pl
...
-  pg = placement_group([bundle1, bundle2], strategy="STRICT_PACK")
+      // Initialize Ray.
+      Ray.init();
+
+      // Construct a list of bundles.
+      List<Map<String, Double>> bundles = new ArrayList<>();
+      Map<String, Double> bundle = new HashMap<>();
+      bundle.put("CPU", 1.0);
+
+      bundles.add(bundle);
```
[the remark this answers is not in the document]
Comment (clay4444) [reply doc/source/placement-group.rst]: updated.

File: doc/source/placement-group.rst (hunk 1 above)
Comment (clay4444) [reply doc/source/placement-group.rst]: updated.

**Pull request merged**